sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PointWriter.scala

```diff
@@ -67,8 +64,15 @@ class PointWriter(container: CosmosAsyncContainer, cosmosWriteConfig: CosmosWrit
   override def scheduleWrite(partitionKeyValue: PartitionKey, objectNode: ObjectNode): Unit = {
     checkState(!closed.get())
 
+    val etag = getETag(objectNode)
+
     cosmosWriteConfig.itemWriteStrategy match {
       case ItemWriteStrategy.ItemOverwrite => upsertWithRetryAsync(partitionKeyValue, objectNode)
+      case ItemWriteStrategy.ItemOverwriteIfNotModified =>
```

Comment:
It won't - because it would recreate documents if they have been deleted after taking the etag snapshot

Reply:
Also would do unconditional replace if no document existed when the snapshot was taken (and etag is empty) but now a document exists - let's say with etag 1 - because we don't have an etag in the df we would effectively do an unconditional upsert (which would overwrite the document someone else created after taking the snapshot)

Reply:
But yes - technically it is possible to use etag with upsert - just doesn't make much sense in most cases for the reasons above

Reply:
Got it, so the main difference is the handling of using etag if the item does not exists